keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ImageResolver.kt

```diff
@@ -46,6 +46,10 @@ class ImageResolver(
         } else {
           artifact.name
         }
+
+        // image names have _ instead of ~
+        artifactName.replace("~", "_")
```

Comment:
That's not going to reassign the value, it returns the amended value. I think you should chain ```.let { it.replace("~", "_") }``` after the if/else.

Reply:
🤦‍♀ lol.

I'd prefer not to chain `.let` after an if/else - I think it's pretty confusing to read. Are you ok with just reassigning the variable?

Reply:
Otherwise, I can just create a function that does the figuring out of the artifact name and pull the logic there.

Reply:
Yeah, a function sounds even better

Reply:
Ok, updated.